### PR TITLE
feat(ST-120): fetch messages for specific Classroom

### PR DESCRIPTION
**Pull request by akib-raihan-sazim**
## Description of Change
- Created MessageCard component to display a message.
- MessageList is created to display all the MessageCard.
- Updated the message.api file for fetching messages from the database
- Used tags ["Message"] for updating the messages without reloads
- Updated the interface IMessage and necessary styles files.
- Interface and styles files are created for all the components

## Associated Ticket Link(s)

- https://trello.com/c/7CwboRQN

## Screenshots / Screen Recordings

https://github.com/user-attachments/assets/469876f0-b6f2-44dd-9a26-0a7acc123521

Screen Shot 2024-09-13 at 12 21 49 AM

### CODE REVIEW

File: shared/redux/rtk-apis/messages/messages.api.ts [hunk 1]
```diff
@@ -9,9 +9,18 @@ const messagesApi = projectApi.injectEndpoints({
         method: "POST",
         body: messageData,
       }),
+      invalidatesTags: ["Messages"],
+    }),
+
+    getClassroomMessages: builder.query<IMessage[], number>({
+      query: (classroomId) => ({
+        url: `/messages/classroom/${classroomId}`,
```
Comment (ishmam-iqbal-sazim) [inline shared/redux/rtk-apis/messages/messages.api.ts:17]: Please follow REST principles. The endpoint should be: `classroooms/:classroomId/messages`

File: modules/Classroom/components/MessageList/MessageList.tsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+import dayjs from "dayjs";
+
+import { useGetClassroomMessagesQuery } from "@/shared/redux/rtk-apis/messages/messages.api";
+
+import MessageCard from "../MessageCard/MessageCard";
+import { IMessageListProps } from "./MessageList.interface";
+import useStyles from "./MessageList.styles";
+
+const MessageList: React.FC<IMessageListProps> = ({ classroomId }) => {
+  const { data: messages } = useGetClassroomMessagesQuery(classroomId);
```
Comment (ishmam-iqbal-sazim) [inline modules/Classroom/components/MessageList/MessageList.tsx:11]: We should also handle loading state in this component.

File: shared/redux/rtk-apis/enrollments/enrollments.types.ts [hunk 3]
```diff
@@ -3,6 +3,7 @@ export interface IUser {
   firstName: string;
   lastName: string;
   email: string;
+  userType: string;
```
Comment (ishmam-iqbal-sazim) [inline shared/redux/rtk-apis/enrollments/enrollments.types.ts:6]: Is this needed for user role?

Comment (akib-raihan-sazim) [reply shared/redux/rtk-apis/enrollments/enrollments.types.ts:6]: Yes bhaiya.

Comment (ishmam-iqbal-sazim) [reply shared/redux/rtk-apis/enrollments/enrollments.types.ts:6]: In that case, could you specify the roles in the type?

**Review by ishmam-iqbal-sazim — APPROVED** [review]
Left a comment

**Pull request merged**